### DISCUSSION TITLE
Unspecified PETSC build.

### DISCUSSION THREAD

**Discussion by Kgeorge1729** [Chaste/Chaste#149] [Q&A] [answered]
Does anyone know how to specify the Petsc installation when generating the makefile for Chaste. When I use ccmake, I keep getting the error, "The pair PETSC_DIR=/usr/lib/petscdir/petsc3.12/x86_64-linux-gnu-real   PETSC_ARCH=x86_64 do not specify a valid PETSc installation." I've tried to change the PETSC_ARCH variable to; linux-intel, x86_64-linux, x86_64-linux-gnu-real, etc. Has anyone else had this issue and how did you solve it?

I'm using Ubuntu 20.04, I'm trying to build the most recent version of Chaste on the develop branch, and this is occurring on a build that I recently updated. I stopped using Chaste for a while and then came back to it and pulled any new commits before trying to build chaste again. Before trying to build Chaste, I installed petsc-dev, I don't know if this might have something to do with it.

**Comment by kwabenantim** [comment]
Hi @Kgeorge1729. This might be worth trying:
`PETSC_DIR=/usr/lib/petscdir/petsc3.12`
`PETSC_ARCH=x86_64-linux-gnu-real`

**Reply by Kgeorge1729** [reply]
Thankyou for the suggestion but this didn't work either unfortunately

**Comment by fcooper8472** [comment, marked as answer]
@Kgeorge1729 can I double check that you have installed the full chaste-dependencies package?

Please make sure you're up-to-date with these instructions:
https://chaste.github.io/docs/installguides/ubuntu-package/

In particular, ensure you have the recommended development dependencies.

```
sudo apt update
sudo apt install --install-recommends chaste-dependencies
```

Assuming you have, there should be no need to specify the PETSc location directly. Please try from a completely clean build directory.
From the main Chaste directory, can you try:

```
mkdir new_build_dir
cd new_build_dir
cmake ..
```

And if that fails in the same way, please could you copy the full output of that cmake command to this discussion.

**Reply by Kgeorge1729** [reply]
Thankyou @fcooper8472 , your suggestion did allow cmake to recognize the petsc build I currently have on my computer.
